core/crypto/client_impl.go

```diff
@@ -37,74 +37,91 @@ type clientImpl struct {
+        keys = append(keys, k)
+    } 
+    
 	// Create Transaction
-	return client.newChaincodeDeployUsingTCert(chaincodeDeploymentSpec, uuid, tCert, nil)
+	return client.newChaincodeDeployUsingTCert(chaincodeDeploymentSpec, uuid, keys, tCert.tCert, nil)
 }
 
 // GetNextTCert Gets next available (not yet used) transaction certificate.
-func (client *clientImpl) GetNextTCert() (tCert, error) {
+func (client *clientImpl) GetNextTCert(attributes map[string]string) (tCert, error) {
 	// Verify that the client is initialized
 	if !client.isInitialized {
 		return nil, utils.ErrNotInitialized
 	}
 
 	// Get next available (not yet used) transaction certificate
-	tCert, err := client.tCertPool.GetNextTCert()
+	tCert, err := client.tCertPool.GetNextTCert(attributes)
 	if err != nil {
 		client.error("Failed getting next transaction certificate [%s].", err.Error())
 		return nil, err
 	}
 
-	return tCert, err
+	return tCert.tCert, err
 }
 
 // NewChaincodeInvokeTransaction is used to invoke chaincode's functions.
-func (client *clientImpl) NewChaincodeExecute(chaincodeInvocation *obc.ChaincodeInvocationSpec, uuid string) (*obc.Transaction, error) {
+func (client *clientImpl) NewChaincodeExecute(chaincodeInvocation *obc.ChaincodeInvocationSpec, attributes map[string]string, uuid string) (*obc.Transaction, error) {
 	// Verify that the client is initialized
 	if !client.isInitialized {
 		return nil, utils.ErrNotInitialized
 	}
 
 	// Get next available (not yet used) transaction certificate
-	tCertHandler, err := client.tCertPool.GetNextTCert()
+	tCertHandler, err := client.tCertPool.GetNextTCert(attributes)
 	if err != nil {
 		client.error("Failed getting next transaction certificate [%s].", err.Error())
 		return nil, err
 	}
 
+	
+    keys := make([]string, len(attributes))
+    for k := range attributes {
+        keys = append(keys, k)
+    } 
+    
 	// Create Transaction
-	return client.newChaincodeExecuteUsingTCert(chaincodeInvocation, uuid, tCertHandler, nil)
+	return client.newChaincodeExecuteUsingTCert(chaincodeInvocation, uuid, keys, tCertHandler.tCert, nil)
 }
 
 // NewChaincodeQuery is used to query chaincode's functions.
-func (client *clientImpl) NewChaincodeQuery(chaincodeInvocation *obc.ChaincodeInvocationSpec, uuid string) (*obc.Transaction, error) {
+func (client *clientImpl) NewChaincodeQuery(chaincodeInvocation *obc.ChaincodeInvocationSpec,  attributes map[string]string, uuid string) (*obc.Transaction, error) {
 	// Verify that the client is initialized
 	if !client.isInitialized {
 		return nil, utils.ErrNotInitialized
 	}
 
 	// Get next available (not yet used) transaction certificate
-	tCertHandler, err := client.tCertPool.GetNextTCert()
+	tCertHandler, err := client.tCertPool.GetNextTCert(attributes)
 	if err != nil {
 		client.error("Failed getting next transaction certificate [%s].", err.Error())
 		return nil, err
 	}
 
+      keys := make([]string, len(attributes))
+    for k := range attributes {
```

Comment:
seems like goimports (which inclides gofmt) needs to be run on this file. Formatting looks off.